src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -67,6 +67,26 @@ public static void PrivateSetter()
             Assert.Null(objCopy.GetMyString());
         }
 
+        [Fact]
+        public static void PrivateSetterPublicGetter()
+        {
+            // https://github.com/dotnet/corefx/issues/37567
+            ClassWithPublicGetterAndPrivateSetter obj
+                = JsonSerializer.Parse<ClassWithPublicGetterAndPrivateSetter>(@"{ ""Class"": {} }");
```

Comment:
Ideally, we should be testing all overloads for these test casess (UTF-8 bytes, async stream, etc.) and same with the serializer side, but that's outside the scope of this PR. cc @steveharter